ui/src/app/applications/components/application-parameters/application-parameters.tsx

```diff
@@ -344,3 +374,16 @@ export const ApplicationParameters = (props: {
         />
     );
 };
+
+// concatMaps merges two maps. Later args take precedence where there's a key conflict.
+function concatMaps(...maps: (Map<string, string> | null)[]): Map<string, string> {
```

Comment:
is there a `util` this could be put into? seems like a common operation

Reply:
Good idea! Done and added test.